src/wallet/tx_builder.rs

```diff
@@ -347,6 +316,80 @@ impl<D: Database, Cs: CoinSelectionAlgorithm<D>> TxBuilder<D, Cs> {
     }
 }
 
+// methods supported only by create_tx
+impl<D: Database> TxBuilder<D, DefaultCoinSelectionAlgorithm, CreateTx> {
```

Comment:
❓ why is this not generic over any coin selection algorithm (it was like this before I think but I only just noticed it)?

Reply:
This is actually a mistake, the only methods that should not be generalized are `with_recipients` and `new`.

The reason why I did that originally was to let the compiler infer which algorithm you want to use without having to manually specify it every time. So if you create the `TxBuilder` with `new()` or `with_recipients()` it immediately gets assigned the `DefaultCoinSelectionAlgorithm`, which you can then override if you want to. But if you don't do it, the compiler can still resolve that generic type to whatever `DefaultCoinSelectionAlgorithm` points to.

Reply:
You can give a default to a struct like `struct Foo<B=Bar>`, perhaps that's what you want here?

Reply:
That would also work I guess. Right now we define a type alias `DefaultCoinSelectionAlgorithm` which can be assigned to whatever type we want to have as default. I would keep it this way because most of the tests are actually designed to work with `LargestFirstCoinSelection`, so I was planning to use the `#[cfg(...)]` attributes to define two different defaults based on whether we are in test mode or not. I would keep largest-first for the tests, and switch to branch and bound for everything else once that's ready.